test/integration/instances_test.go

```diff
@@ -589,6 +589,211 @@ func TestInstance_withPG(t *testing.T) {
+	_, instanceTeardown, err := createInstanceWithReservedIP(t, client, reservedIP.Address)
+	if err != nil {
+		t.Fatalf("Failed to create initial instance: %v", err)
+	}
+	defer instanceTeardown()
+
+	// Now try to create another instance with the same IP
+	_, secondInstanceTeardown, err := createInstanceWithReservedIP(t, client, reservedIP.Address)
+	if err == nil {
+		t.Errorf("Expected error with already assigned reserved IP, but got none")
+		secondInstanceTeardown()
+	}
+}
+
+func TestInstance_CreateWithNonReservedAddress(t *testing.T) {
+	client, teardown := createTestClient(t, "fixtures/TestInstance_CreateWithNonReservedAddress")
+	defer teardown()
+
+	_, instanceTeardown, err := createInstanceWithReservedIP(t, client, "192.0.2.1")
+	if err == nil {
+		t.Errorf("Expected error with non-reserved address, but got none")
+		instanceTeardown()
+	}
+}
+
+func TestInstance_CreateWithNonOwnedReservedAddress(t *testing.T) {
+	client, teardown := createTestClient(t, "fixtures/TestInstance_CreateWithNonOwnedReservedAddress")
+	defer teardown()
+
+	_, instanceTeardown, err := createInstanceWithReservedIP(t, client, "198.51.100.1")
+	if err == nil {
+		t.Errorf("Expected error with non-owned reserved address, but got none")
+		instanceTeardown()
+	}
+}
+
+func TestInstance_CreateWithEmptyIPAddress(t *testing.T) {
+	client, teardown := createTestClient(t, "fixtures/TestInstance_CreateWithEmptyIPAddress")
+	defer teardown()
+
+	_, instanceTeardown, err := createInstanceWithReservedIP(t, client, "")
+	if err == nil {
+		t.Errorf("Expected error with empty IP address, but got none")
+		instanceTeardown()
+	}
+}
+
+func TestInstance_CreateWithNullIPAddress(t *testing.T) {
+	client, teardown := createTestClient(t, "fixtures/TestInstance_CreateWithNullIPAddress")
+	defer teardown()
+
+	_, instanceTeardown, err := createInstanceWithReservedIP(t, client, "", func(client *linodego.Client, opts *linodego.InstanceCreateOptions) {
+		opts.Ipv4 = nil
+	})
+	if err != nil {
+		t.Errorf("Unexpected error with null IP address: %v", err)
+	} else {
+		instanceTeardown()
+	}
+}
+
+func TestInstance_CreateWithMultipleIPAddresses(t *testing.T) {
+	client, teardown := createTestClient(t, "fixtures/TestInstance_CreateWithMultipleIPAddresses")
+	defer teardown()
+
+	reservedIP, err := client.ReserveIPAddress(context.Background(), linodego.ReserveIPOptions{Region: "us-east"})
+	if err != nil {
+		t.Fatalf("Failed to reserve IP: %v", err)
+	}
+	defer func() {
+		err := client.DeleteReservedIPAddress(context.Background(), reservedIP.Address)
+		if err != nil {
+			t.Errorf("Failed to delete reserved IP: %v", err)
+		}
+	}()
+
+	_, instanceTeardown, err := createInstanceWithReservedIP(t, client, "", func(client *linodego.Client, opts *linodego.InstanceCreateOptions) {
+		opts.Ipv4 = []string{reservedIP.Address, "192.0.2.2"}
+	})
+	if err == nil {
+		t.Errorf("Expected error with multiple IP addresses, but got none")
+		instanceTeardown()
```

Comment:
Same as above move it right after createInstanceWithReservedIP  and before the error checks. Also, use defer

Reply:
Added defer statement immediately after the creation statement.